docs/training-run/index.txt

```diff
@@ -4,106 +4,255 @@
+training class
+
+Custom Script
+-------------
+
+.. code:: bash
+
+   script.py [args...]
+
+A custom script can be launched under supported launch layers instead of a trial class definition,
+with arguments passed as expected.
+
+Horovod Launcher
+================
+
+Format:
+
+``determined.launch.horovod [[HVD_OVERRIDES...] --] (--trial TRIAL)|(SCRIPT...)``
+
+The module accepts arguments to be passed directly to the Horovod launcher called under-the-hood,
+horovodrun, which overrides automatically set values. See the official Horovod documentation for
+more information about the horovodrun launcher. The optional override arguments must end with a
+``--`` separator before the trial specification.
+
+Example:
+
+.. code:: bash
+
+   python3 -m determined.launch.horovod --fusion-threshold-mb 1 --cycle-time-ms 2 -- --trial model_def:MyTrial
+
+PyTorch Distributed Launcher
+============================
+
+Format:
+
+``determined.launch.torch_distributed [[TORCH_OVERRIDES...] --] (--trial TRIAL)|(SCRIPT...)``
+
+This launcher is a Determined wrapper around PyTorch's native distributed training launcher,
+torch.distributed.run. Any arbitrary override arguments to torch.distributed.run are accepted, which
+overrides default values set by Determined. See the official PyTorch documentation for information
+about how to use ``torch.distributed.run``. The optional override arguments must end with a ``--``
+separator before the trial specification.
 
-The ``det experiment create`` command requires two arguments:
+Example:
 
--  The :ref:`configuration file <experiment-configuration>` is a :ref:`YAML <topic-guides_yaml>`
-   file that configures the experiment.
--  The context directory contains all code that are relevant to training and will be uploaded to the
-   master.
+.. code:: bash
 
-We don't allow the total size of the files in the context to exceed 95 MiB. As a result, datasets
-should typically not be included unless they are very small; instead, users should set up data
-loaders to read data from an external source. Refer to :ref:`preparing data <prepare-data>` for more
-suggestions on data loading.
+   python3 -m determined.launch.torch_distributed --rdzv_endpoint=$CUSTOM_RDZV_ADDR -- --trial model_def:MyTrial
 
-Since project directories might include large artifacts that should not be packaged as part of the
-model definition (e.g., data sets or compiled binaries), users can optionally include a
-``.detignore`` file at the top level that specifies file paths to be omitted from the model
-definition. The ``.detignore`` file uses the same syntax as `.gitignore
-<https://git-scm.com/docs/gitignore>`__. Note that byte-compiled Python files (e.g., ``.pyc`` files
-or ``__pycache__`` directories) are always ignored.
+DeepSpeed Launcher
+==================
 
-Local Test Mode
+Format:
+
+``determined.launch.deepspeed [[DEEPSPEED_ARGS...] --] (--trial TRIAL)|(SCRIPT...)``
```

Comment:
i believe deepspeed doesn't support additional override args (right @liamcli ?) so maybe we should remove this

Reply:
Not resolved.

Reply:
Done